pkg/controller/service/servicecontroller.go

```diff
@@ -422,10 +422,15 @@ func (s *serviceCache) delete(serviceName string) {
 }
 
 func (s *ServiceController) needsUpdate(oldService *api.Service, newService *api.Service) bool {
-	if !wantsLoadBalancer(oldService) && !wantsLoadBalancer(newService) {
+	oldServiceNeedsLoadBalancer := wantsLoadBalancer(oldService)
```

Comment:
This is going to change the behaviour of service controller to periodically call EnsureLoadBalancer even when there is no change in the service.